### PR TITLE
LOW-3: Missing input validation

### PR DESCRIPTION
**Pull request by alistair-singh**
Resolves: SNO-753

Extra merkle proof validations from the audit issue are not addressed.

### CODE REVIEW

File: contracts/src/BeefyClient.sol [hunk 1]
```diff
@@ -215,6 +216,9 @@ contract BeefyClient {
         ValidatorSet memory _initialValidatorSet,
         ValidatorSet memory _nextValidatorSet
     ) {
+        if (_nextValidatorSet.id != _initialValidatorSet.id + 1) {
+            revert InvalidValidatorSetData();
```
Comment (vgeddes) [inline contracts/src/BeefyClient.sol:220]: Can just `revert()` here without raising a custom exception. I don't want to add a custom exception `InvalidValidatorSetData` which may never ever be used.

Comment (alistair-singh) [reply contracts/src/BeefyClient.sol]: Addressed in https://github.com/Snowfork/snowbridge/pull/1025/commits/9fd119560d062741cc7513e102778fa7d21e5850.

Comment (vgeddes) [reply contracts/src/BeefyClient.sol]: Nit, would prefer a string that looks more robotic: `invalid-constructor-params`

**Review by vgeddes — APPROVED** [review]
+1

**Pull request merged**